### PR TITLE
add small caching for teem and careers

### PR DESCRIPTION
**Pull request by BohdanLeshchyshyn**

### CODE REVIEW

**Comment by epaminond** [comment]
@dmytropanontko , could you review this one?

File: server/index.js [hunk 1]
```diff
@@ -423,12 +423,15 @@ app.prepare().then(() => {
   server.get('/blog/:name', (req, res) => app.render(req, res, '/post', { name: req.params.name }));
 
   server.get('/api/astronauts', async (req, res) => {
-    const team = await getTeam();
+
+    let team = await getTeam();
```
Comment (dmytropanontko) [inline server/index.js]: @BohdanLeshchyshyn Why did you change it? If you don't change variables we must use `const`

File: server/get-info-from-googleapis.js [hunk 2]
```diff
@@ -72,12 +74,13 @@ const getData = async (rangeName, sheetId) => {
     const sheetData = await getSheets(sheetId);
 
     const team = getSheetValues(sheetData.valueRanges, rangeName);
+    cashedTeam = team;
```
Comment (dmytropanontko) [inline server/get-info-from-googleapis.js]: @BohdanLeshchyshyn it'll make a bug) because `getData` method uses for fetching two types of data: `careers` and `team`. That's why we will be cache one-time `team` and other-time `careers`)))

You need to do the following:
1) Rename [team](https://github.com/keenethics/keenethics/blob/de65da77e9d6afb5caf88b9bbd0ce015220c6b7b/server/get-info-from-googleapis.js#L76) to `data`. It was my fail)))
2) and place for caching must be [here](https://github.com/keenethics/keenethics/blob/de65da77e9d6afb5caf88b9bbd0ce015220c6b7b/server/get-info-from-googleapis.js#L90) for `careers` and [here](https://github.com/keenethics/keenethics/blob/de65da77e9d6afb5caf88b9bbd0ce015220c6b7b/server/get-info-from-googleapis.js#L89) for `teams`

File: server/index.js [hunk 3]
```diff
@@ -423,12 +423,15 @@ app.prepare().then(() => {
...
+
     res.send(JSON.stringify(team));
   });
 
   server.get('/api/careers', async (req, res) => {
-    const careers = await getCareers();
+    let careers = await getCareers();
```
Comment (dmytropanontko) [inline server/index.js]: same :point_up:

File: server/index.js [hunk 4]
```diff
@@ -423,12 +423,15 @@ app.prepare().then(() => {
   server.get('/blog/:name', (req, res) => app.render(req, res, '/post', { name: req.params.name }));
 
   server.get('/api/astronauts', async (req, res) => {
-    const team = await getTeam();
+
```
Comment (dmytropanontko) [inline server/index.js]: remove empty line)))

File: server/get-info-from-googleapis.js (hunk 2 above)
Comment (dmytropanontko) [reply server/get-info-from-googleapis.js]: OR you can cache `sheetData` but in this way you need to make bit changes
```js
  } catch (err) {
    console.error(err);
    // TODO: rename cashedTeam to cachedSheetData
    const data = getSheetValues(cashedTeam.valueRanges, rangeName);
    return arrayOfArraysToCollection(data);
  }
```

and this approach better for me

**Comment by dmytropanontko** [comment]
@epaminond @BohdanLeshchyshyn Good job)

File: server/get-info-from-googleapis.js [hunk 5]
```diff
@@ -71,13 +73,16 @@ const getData = async (rangeName, sheetId) => {
   try {
     const sheetData = await getSheets(sheetId);
 
-    const team = getSheetValues(sheetData.valueRanges, rangeName);
+    cashedSheetData = sheetData;
+
+    const data = getSheetValues(sheetData.valueRanges, rangeName);
 
-    return arrayOfArraysToCollection(team);
+    return arrayOfArraysToCollection(data);
```
Comment (epaminond) [inline server/get-info-from-googleapis.js]: Why are you caching an intermediate result, not an end one?

File: server/index.js [hunk 6]
```diff
@@ -424,11 +424,13 @@ app.prepare().then(() => {
 
   server.get('/api/astronauts', async (req, res) => {
     const team = await getTeam();
+
     res.send(JSON.stringify(team));
   });
 
   server.get('/api/careers', async (req, res) => {
     const careers = await getCareers();
+
```
Comment (epaminond) [inline server/index.js]: Are these changes required?

File: server/get-info-from-googleapis.js (hunk 5 above)
Comment (dmytropanontko) [reply server/get-info-from-googleapis.js]: @epaminond It's my recommendation. I have two reasons why do it:
1) `getData` calls every time when a user goes to `careers` and `team`. That's mean our cache will be refreshed.
2) It'll create more code than now)

Comment (epaminond) [reply server/get-info-from-googleapis.js]: > getData calls every time when a user goes to careers and team. That's mean our cache will be refreshed.

So what? I suggest smth like this:

```js
const data = getSheetValues(sheetData.valueRanges, rangeName);
cashedSheetData = arrayOfArraysToCollection(data);
return cashedSheetData;
```

**Comment by epaminond** [comment]
@dmytropanontko , could we deploy this one?

**Comment by dmytropanontko** [comment]
@epaminond deployed)))
